caus.py

```diff
@@ -1,79 +1,77 @@
 import math
-from elasticity import elasticity
+from elasticity import Elasticity
+from typing import Optional, Tuple
+
 class CAUS:
+    """
+    Base class for all kinds of custom autoscalers.
+    """
+
+    def calculate_replicas(self, publishing_rate: float, current_replicas: int) -> Tuple[Optional[int], int]:
+        """
+        Calculate the amount of desired replicas and buffered replicas given the measured publishing rate and the current amount of replicas.
+        """
+        pass
 
-# vars for parameterization of elasticity, commented out in order to be instance variables instead of class variables
-# in order to not have side effect regarding multiple mutable objects
-# elastic capacity: float
-#elasticityCapacity = 0.0
-# elastic min replicas: int or None
-#elasticityMinReplicas = None
-# elastic max replicas: int or None
-#elasticityMaxReplicas = None
-# elastic initial buffer: int
-#elasticityBufferInitial = 0
-# elastic buffered replicas: int
-#elasticityBufferedReplicas = 0
-# elastic buffer threshold: float
-#elasticityBufferThreshold = 70.0
 
-#TODO  push elasticity stuff to own class?
+class SimpleCAUS(CAUS):
 
-    def __init__(self, elasticity):
+    def __init__(self, elasticity: Elasticity):
         self.elasticity = elasticity
 
-    # AdjustBufferAmount adjusts the buffer size depending on the current publishing rate
-    # publishingRate            -  the measured publishing rate
-    # currentCapacity allocated -  this is the total number of allocated instances
-    # currentBuffer             -  this is the current buffer size
-    # currentPerf               -  this is the performance metric
-    # bufferThreshold           -  the threshold to increase the buffer size
-    def adjustBufferAmount(self, publishingRate, currentReplicas, currentBuffer, currentPerf,bufferThreshold,initialBuffer):
-        usage = publishingRate/((currentReplicas - currentBuffer)*currentPerf)
-        bufferThresh = bufferThreshold / 100.0
+    def adjust_buffers(self, publishing_rate: float, current_replicas: int, current_buffers: int, current_metric_performance: float, buffer_threshold: float, initial_buffer: int) -> int:
+        """
+        adjust_buffers returns the new buffer size depending on the current publishing rate
+        publishing_rate            -  the measured publishing rate
```

Comment:
is this a common docstring format?
maybe in the future, it would be nice to have generated documentation using sphinx  
typically, rst format is used, however plugins for other formats exist (I personally prefer the Google style, but choose as you wish)

Reply:
For now, I restricted the documentation changes to converting the previous comments to actual docstrings.
The reasoning is simply the following:
As far as I know, there is no common standard on how to write Python docstrings.
So, you want the google style?
Okay, I can do that.

Reply:
basically any style sphinx can work with is fine for me  
to my knowledge, this is standard rst, google and numpy style, I personally prefer Google, but as I said, take any of those that you like the most (or even a totally different one if there is a plugin which allows us to generate nice docs)

Reply:
Are you sure that will be more used than the java one I already had to remove again?

Reply:
Google style seems to be supported for that: https://opensource.com/article/19/11/document-python-sphinx

Reply:
> Are you sure that will be more used than the java one I already had to remove again?
First, CAUS is meant to be extended with plugins (at least in some point in the future), so I think that code documentation will be more useful compared to most services of t2 which just need to be deployed.
Additionally, python support is better in readthedocs.
So will it be more used? Sadly, I somewhat doubt it. But I still think having it would be nice